framework/src/kernels/Kernel.C

```diff
@@ -80,6 +80,25 @@ Kernel::computeJacobian()
 
   ke += _local_ke;
 
+  if (_calculate_nonlocal_variable_jacobian)
+  {
+    _nonlocal_ke.resize(ke.m(), _subproblem.es().n_dofs());
+    _nonlocal_ke.zero();
+    for (_i = 0; _i < _test.size(); _i++)
```

Comment:
Yeah, again, we are shooting for a first proof of concept to be run with the JacobianAnalyzer on tiny test problems first. We do have a lot more ideas already that go beyond this PR and we are largely aware of the looming challenges (more parallel communication when assembling the Jacobian for example).